File: Domain/Activity.cs
// <copyright file="Activity.cs" company="Jimmy Kurian">
// Copyright (c) Jimmy Kurian. All rights reserved.
// </copyright>

namespace Domain
{
    /// <summary>
    /// Represents an activity with a title, date, and other details.
    /// </summary>
    public class Activity
    {
        /// <summary>
        /// Gets or sets the identifier for the Activity.
        /// </summary>
        public Guid Id { get; set; }

        /// <summary>
        /// Gets or sets the title of the Activity.
        /// </summary>
        public string? Title { get; set; }

        /// <summary>
        /// Gets or sets the date of the Activity.
        /// </summary>
        public DateTime Date { get; set; }

        /// <summary>
        /// Gets or sets the description of the Activity.
        /// </summary>
        public string? Description { get; set; }

        /// <summary>
        /// Gets or sets the category of the Activity.
        /// </summary>
        public string? Category { get; set; }

        /// <summary>
        /// Gets or sets the city where the Activity takes place.
        /// </summary>
        public string? City { get; set; }

        /// <summary>
        /// Gets or sets the venue of the Activity.
        /// </summary>
        public string? Venue { get; set; }
    }
}

File: client-app/src/app/models/activity.ts
/**
 * @author Jimmy Kurian
 * @name Activity
 */

/**
 * The Activity interface represents the structure of an activity object in the Reactivities application.
 * It includes various properties such as id, title, date, description, category, city, and venue.
 *
 * @interface
 *
 * @property {string} id - The unique identifier for the activity.
 * @property {string} title - The title of the activity.
 * @property {string} date - The date of the activity in ISO string format.
 * @property {string} description - A brief description of the activity.
 * @property {string} category - The category to which the activity belongs.
 * @property {string} city - The city where the activity takes place.
 * @property {string} venue - The specific venue of the activity within the city.
 */
export interface Activity {
  id: string;
  title: string;
  date: string;
  description: string;
  category: string;
  city: string;
  venue: string;
}
